_episodes/03-communicating-data.md

```diff
@@ -23,91 +23,257 @@ parallelisation often relies on how you communicate data.
+> >    are 64 bit.
+> > 3. `MPI_BYTE` or `MPI_CHAR` - you may want to use [strlen](https://man7.org/linux/man-pages/man3/strlen.3.html) to
+> >    calculate how many elements of `MPI_CHAR` being sent
 > {: .solution}
 {: .challenge}
 
-### Communication modes
+## Communicators
 
-There are multiple "modes" on how data is sent in MPI: standard, buffered, synchronous and ready. When an MPI
-communication function is called, control/execution of the program is passed from the calling program to the MPI
-function. Your program won't continue until MPI is happy that the communication happened successfully. The difference
-between the communication modes is the criteria of a successful communication.
+All communication in MPI is handled by something known as a *communicator*. We can think of a communicator as being a
+collection of ranks which are able to exchange data with one another. What this means is that every communication
+between two (or more) ranks is linked to a specific communicator in the program. When we run an MPI application, the
+ranks will belong to the default communicator known as `MPI_COMM_WORLD`. We've seen this in earlier episodes when, for
+example, we've used functions like `MPI_Comm_rank()` to get the rank number,
 
-To use the different modes, we don't pass a special flag. Instead, MPI uses different functions to separate the
-different modes. The table below lists the four modes with a description and their associated functions (which will be
-covered in detail in the following episodes).
+```c
+int my_rank;
+MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);  /* MPI_COMM_WORLD is the communicator the rank belongs to */
+```
 
-| Mode | Description | MPI Function |
-| - | - | - |
-| Synchronous | Returns control to the program when the message has been sent and received successfully | `MPI_Ssend()` |
-| Buffered | Control is returned when the data has been copied into in the send buffer, regardless of the receive being completed or not | `MPI_Bsend()` |
-| Standard  | Either buffered or synchronous, depending on the size of the data being sent and what your specific MPI implementation chooses to use | `MPI_Send()` |
-| Ready | Will only return control if the receiving rank is already listening for a message | `MPI_Rsend()` |
+In addition to `MPI_COMM_WORLD`, we can make sub-communicators and distribute ranks into them. Messages can only be
+sent and received to and from the same communicator, effectively isolating messages to a communicator. For most
+applications, we usually don't need anything other than `MPI_COMM_WORLD`. But organising ranks into communicators can be
+helpful in some circumstances, as you can create small "work units" of multiple ranks to dynamically schedule the
+workload, or to help compartmentalise the problem into smaller chunks by using a [virtual cartesian
+topology](https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node192.htm#Node192). Throughout this lesson, we will

```

Comment:
Could we not say "throughout this course" instead? (At least, as long as we don't use any other type of communicator?)